src/runtime-rs/crates/resource/src/volume/utils.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{fs, path::Path};
+
+use anyhow::{anyhow, Context, Result};
+
+use crate::{
+    share_fs::{do_get_guest_path, do_get_host_path},
+    volume::share_fs_volume::generate_mount_path,
+};
+use kata_sys_util::eother;
+use kata_types::mount::{get_volume_mount_info, DirectVolumeMountInfo};
+
+pub const DEFAULT_VOLUME_FS_TYPE: &str = "ext4";
+pub const KATA_MOUNT_BIND_TYPE: &str = "bind";
+pub const KATA_DIRECT_VOLUME_TYPE: &str = "directvol";
+pub const KATA_VFIO_VOLUME_TYPE: &str = "vfiovol";
+pub const KATA_SPDK_VOLUME_TYPE: &str = "spdkvol";
+
+// volume mount info load infomation from mountinfo.json
+pub fn volume_mount_info(volume_path: &str) -> Result<DirectVolumeMountInfo> {
+    get_volume_mount_info(volume_path)
+}
+
+pub fn make_path_exist_ok(path: &Path) -> Result<()> {
```

Comment:
What is the advantage of using `fs::create_dir_all`?

Reply:
fixed. it'll be deleted. no one will call it.